### ISSUE TITLE
Overload resolution problem

### ISSUE THREAD

**Issue by HurryStarfish** [bmx-ng/bcc#291]
When a function or method has two overloaded versions that differ only in the type of one of the parameters, and the parameter type of one version extends/implements that of the other version, NG should pick the closest fitting version to call. For example:
```
SuperStrict
Framework BRL.basic

Local o1:TTest1 = New TTest1
Local o2:TTest2 = New TTest2

F o1
F o2

Type TTest1                End Type
Type TTest2 Extends TTest1 End Type

Function F(o:TTest1) Print "t1" End Function
Function F(o:TTest2) Print "t2" End Function
```
This currently prints 

> t1
> t1

If the two function declarations are swapped, it does not even compile, instead outputting the error
> Compile Error: Duplicate declaration Function F(o:TTest1)

In both cases (no matter which order the functions are in) the expected correct result would be:

> t1
> t2

The above does not only apply to (class/subclass) overload pairs, but also to (interface/subinterface) and (interface/implementing class).

**Comment by GWRon** [comment]
:+1: 

PS: seems you have a run to find bugs today. Keeps Brucey busy the next days.

**Comment by HurryStarfish** [comment]
Yep, I've been looking into the new generics feature, and trying to write a linked list implementation for BRL.Collections (well, to "port" BRL.List actually) and these are the things I stumbled upon in the process. I got a bunch more things to say, especially about BRL.Collections, but reporting the bugs comes first :)
